src/shared/openssl-util.c

```diff
@@ -1125,6 +1125,83 @@ int digest_and_sign(
         return 0;
 }
 
+int pkcs7_new(X509 *certificate, EVP_PKEY *private_key, PKCS7 **ret_p7, PKCS7_SIGNER_INFO **ret_si) {
```

Comment:
this is very opaque. please add a comment explaining what this does, because it's not obvious to me at all

Reply:
maybe mention explicitly that private_key is optional and what it means to specify it/not to specify it